src/components/charts/LoadingOverlay.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <div class="loading-overlay" :style="style">
+    <v-icon size="x-large">mdi-loading mdi-spin</v-icon>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+
+interface Props {
+  margin?: {
+    top?: number
+    right?: number
+    bottom?: number
+    left?: number
+  }
+}
+
+const props = defineProps<Props>()
+
+const style = computed(() => ({
+  paddingTop: props.margin?.top ? `${props.margin.top}px` : undefined,
```

Comment:
It looks a bit weird that you're setting the padding with a property called `margin`. Maybe add a comment why this is the case?